internal/services/containers/container_registry_cache_rule_resource.go

```diff
@@ -0,0 +1,252 @@
+			ForceNew:    true,
+			Description: "The full source repository path such as 'docker.io/library/ubuntu'.",
+		},
+
+		"target_repo": {
+			Type:        pluginsdk.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			Description: "The target repository namespace such as 'ubuntu'.",
+		},
+	}
+}
+
+func (ContainerRegistryCacheRule) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (ContainerRegistryCacheRule) ModelObject() interface{} {
+	return nil
+}
+
+func (ContainerRegistryCacheRule) ResourceType() string {
+	return "azurerm_container_registry_cache_rule"
+}
+
+func (r ContainerRegistryCacheRule) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			cacheRulesClient := metadata.Client.Containers.ContainerRegistryClient_v2023_07_01.CacheRules
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			ctx, cancel := timeouts.ForCreate(metadata.Client.StopContext, metadata.ResourceData)
+
+			defer cancel()
+			log.Printf("[INFO] preparing arguments for Container Registry Cache Rule creation.")
+
+			registryId, err := registries.ParseRegistryID(metadata.ResourceData.Get("container_registry_id").(string))
+			if err != nil {
+				return err
+			}
+
+			resourceGroup := registryId.ResourceGroupName
+			registryName := registryId.RegistryName
+
+			id := cacherules.NewCacheRuleID(subscriptionId,
+				resourceGroup,
+				registryName,
+				metadata.ResourceData.Get("name").(string),
+			)
+
+			if metadata.ResourceData.IsNewResource() {
+				existing, err := cacheRulesClient.Get(ctx, id)
+
+				if err != nil {
+					if !response.WasNotFound(existing.HttpResponse) {
+						return fmt.Errorf("checking for presence of existing %s: %s", id, err)
+					}
+				}
+
+				if !response.WasNotFound(existing.HttpResponse) {
+					return tf.ImportAsExistsError("azurerm_container_registry_cache_rule", id.ID())
+				}
+
+				// TODO: make a check that the repo is available in the registry.
+				targetRepo := metadata.ResourceData.Get("target_repo").(string)
+
+				// TODO: validate the source repo.
+				sourceRepo := metadata.ResourceData.Get("source_repo").(string)
+
+				credentialSetId := metadata.ResourceData.Get("credential_set_id").(string)
+
+				parameters := cacherules.CacheRule{
+					Name: &id.CacheRuleName,
+					Properties: &cacherules.CacheRuleProperties{
+						SourceRepository: &sourceRepo,
+						TargetRepository: &targetRepo,
+					},
+				}
+
+				// Conditionally add CredentialSetResourceId if credentialSetId is not empty
+				if credentialSetId != "" {
+					parameters.Properties.CredentialSetResourceId = &credentialSetId
+				}
+
+				if err := cacheRulesClient.CreateThenPoll(ctx, id, parameters); err != nil {
+					return fmt.Errorf("creating Container Registry Cache Rule %s: %+v", id, err)
+				}
+			}
+
+			metadata.SetID(id)
+
+			return r.Read().Func(ctx, metadata)
```

Comment:
Typed resources such as this one will automatically call the read function after the create so we don't need to do this explicitly 
```suggestion
			return nil
```